src/modules/sbstudio/plugin/gateway.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from functools import lru_cache
+from typing import TypeVar
+
+from sbstudio.api import SkybrushGatewayAPI
+from sbstudio.api.errors import NoOnlineAccessAllowedError, SkybrushStudioAPIError
+
+__all__ = ("get_gateway",)
+
+T = TypeVar("T")
+
+#############################################################################
+# configure logger
+
+log = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=1)
+def _get_gateway_from_url(url: str):
+    """Constructs a Skybrush Gateway API object from a root URL.
+
+    Memoized so we do not need to re-construct the same instance as long as
+    the user does not change the add-on settings.
+
+    Raises:
+        ValueError on initialization error
```

Comment:
```suggestion
        ValueError: on initialization error
```

(this is the standard syntax that can be parsed)